flow/raster_cache.cc

```diff
@@ -39,8 +46,29 @@ void RasterCacheResult::draw(DlCanvas& canvas, const DlPaint* paint) const {
              std::abs(bounds.height() - image_->dimensions().height()) <= 1);
   canvas.TransformReset();
   flow_.Step();
-  canvas.DrawImage(image_, {bounds.fLeft, bounds.fTop},
-                   DlImageSampling::kNearestNeighbor, paint);
+  if (is_root_canvas || !rtree_) {
+    canvas.DrawImage(image_, {bounds.fLeft, bounds.fTop},
+                     DlImageSampling::kNearestNeighbor, paint);
+  } else {
+    // On some platforms RTree from overlay layers is used for unobstructed
+    // platform views and hit testing. To preserve the RTree raster cache must
+    // paint individual rects instead of the whole image.
+    auto rects = rtree_->searchAndConsolidateRects(kGiantRect);
+
+    // Bounds of original display list.
+    SkRect rtree_bounds = SkRect::MakeEmpty();
```

Comment:
rtree should probably have an overall bounds which would make this unnecessary. It can be retrieved for free from the top-level node which bounds the entire tree.

Reply:
I added RTree::bounds() implementation: 
```cc
const SkRect& DlRTree::bounds() const {
  if (!nodes_.empty()) {
    return nodes_.back().bounds;
  } else {
    return empty_;
  }
}
``` 

This should be correct, right? Unless I'm misunderstanding how the R-Tree works.

Reply:
Absolutely!